packages/cli/generation/local-generation/local-workspace-runner/src/runLocalGenerationForWorkspace.ts

```diff
@@ -119,7 +125,22 @@ async function writeFilesToDiskAndRunGenerator({
     if (firstLocalOutputItem == null) {
         return;
     }
-    await rm(absolutePathToLocalOutput, { force: true, recursive: true });
+    // await rm(absolutePathToLocalOutput, { force: true, recursive: true });
+    const dotFernIgnoreFile = path.join(absolutePathToLocalOutput, ".fernignore");
+    let filesToBeIgnoredByFern: string[] = [".fernignore"];
+    const dotFernIgnoreFileExists = existsSync(dotFernIgnoreFile);
+
+    if (dotFernIgnoreFileExists) {
+        const dotFernIgnoreFileContent = await readFile(dotFernIgnoreFile, "utf-8");
+        filesToBeIgnoredByFern = filesToBeIgnoredByFern.concat(dotFernIgnoreFileContent.trim().split(/\r?\n/));
+        filesToBeIgnoredByFern = filesToBeIgnoredByFern.filter(filePattern => filePattern !== "" && !filePattern.startsWith("#"));
+        filesToBeIgnoredByFern = globSync(filesToBeIgnoredByFern, {
+            cwd: absolutePathToLocalOutput, root: "", dot: true, nobrace: false, noext: false, matchBase: true
+        });
+        await EXEC("git init && git add . && git commit -m \"initial\" && git rm -rf .", {cwd: absolutePathToLocalOutput});
```

Comment:
nit: can we run each command separately and `await` it. itll be better to handle errors that way as well. Why do we need the `git commit -m` ?

Reply:
shouldn't all of these commands also be run after the cp on line 144-151

Reply:
> nit: can we run each command separately and `await` it. itll be better to handle errors that way as well. Why do we need the `git commit -m` ?

Of course, I’ll separate them. Without the ‘git commit -m’, ‘git restore .’ was returning.

Reply:
ah interesting

Reply:
> shouldn't all of these commands also be run after the cp on line 144-151

I haven’t figured. I’ll sure look into this as well.

Reply:
Thank You @dsinghvi for the feedbacks and suggestions.